### DISCUSSION TITLE
Rendering from code.

### DISCUSSION THREAD

**Discussion by Throndax57** [AdamsLair/duality#887] [Questions & Help] [answered]
I'm trying to make a dialogue box system on a game I'm making and I plan on having a spritesheet with letters as the sprite. The problem is I don't know how or if its even possible to render/create a sprite from code.

If you have a better idea for the dialouge box, please do tell me.

**Comment by SirePi** [comment, marked as answer]
If you are using v3 (downloaded from duality's page), you could download YAUI from the Package Manager (bit of explanations on how it works [here](https://github.com/SirePi/duality-ui/wiki) ); if you are already trying the v4 alpha, you can probably grab it from nuGet.

Otherwise, you can take a look at the `Canvas` class [here](https://docs.duality2d.net/pages/v3/Custom-Renderers.html#using-the-canvas-class), and especially at the `DrawText` method.
